### PR TITLE
fix adp related code

### PR DESCRIPTION
**Pull request by pinzart**
### Purpose

Fix build.
Some analytics tests fail due to code left outside of DebugMode scope

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCore/Logging/DynamoAnalyticsClient.cs [hunk 1]
```diff
@@ -292,8 +295,6 @@ public void TrackScreenView(string viewName)
 
         public void TrackException(Exception ex, bool isFatal)
         {
-            if (!ReportingAnalytics) return;
-
```
Comment (pinzart) [inline src/DynamoCore/Logging/DynamoAnalyticsClient.cs:296]: @QilongTang do you know if this is intentional ?
Exception tracking seems to have been enabled regardless of Analytics Opt-in status.

Comment (pinzart) [reply src/DynamoCore/Logging/DynamoAnalyticsClient.cs:296]: It is even covered by a test...

Comment (QilongTang) [reply src/DynamoCore/Logging/DynamoAnalyticsClient.cs:296]: @pinzart Wait doesn't think mean if not reporting then simply return here?

Comment (pinzart) [reply src/DynamoCore/Logging/DynamoAnalyticsClient.cs:296]: Yes. (the code that I added, as part of ADP work).
The code before my changes did not have that condition...so it reported exceptions even if the user was opted out

Comment (pinzart) [reply src/DynamoCore/Logging/DynamoAnalyticsClient.cs:296]: I want to know if that was a feature request ... or if I can leave my changes and update the existing test..

**Comment by pinzart** [comment]
analytics test are passing locally 
@QilongTang good to merge ?

**Comment by QilongTang** [comment]
@pinzart Good to merge

**Pull request merged**